vllm/engine/output_processor/multi_step.py

```diff
@@ -72,6 +79,11 @@ def process_outputs(self, sequence_group: SequenceGroup,
         seqs = sequence_group.get_seqs(status=SequenceStatus.RUNNING)
 
         assert seqs, "expected running sequences"
+
+        if len(seqs) > 1 or sequence_group.sampling_params.best_of > 1:
```

Comment:
add a comment here explaining what's going on. also can we explicitly fail if beam search is enabled?

Reply:
done. 

re: failing on beam search, I think this is best done *before* we put the requests in the batch so I've added some code in `add_request` to that effect.